app/src/main/java/com/duckduckgo/app/browser/downloader/NetworkFileDownloader.kt

```diff
@@ -34,8 +38,50 @@ class NetworkFileDownloader @Inject constructor(private val context: Context, pr
             return
         }
 
-        val guessedFileName = filenameExtractor.extract(pendingDownload)
+        fileService.getFileDetails(pendingDownload.url)?.enqueue(object : Callback<Void> {
+            override fun onResponse(call: Call<Void>, response: Response<Void>) {
+                if (response.isSuccessful) {
+                    var updatedPendingDownload = pendingDownload.copy()
 
+                    if (response.headers().get("content-type") != null) {
```

Comment:
Following @aitorvs's comment, not sure if we can avoid doing multiple copies but it would be nice.

Is there a case where any of the headers could come back as null but `PendingFileDownload` already had a `mimeType` or `contentDisposition` value set? If the answer is no we can set those two parameters even though they come back as null. If we don't know the answer or we think is a possible edge case, we can keep this as it is.

Reply:
Not anymore, we use this call to populate the `mimeType` and `contentDisposition` of  `PendingFileDownload`. If the headers come back as null, DownloadManager won't be able to download the file, therefore we will eventually call `callback.downloadFailed()`